rotkehlchen/tests/data_migrations/test_migrations.py

```diff
@@ -235,3 +249,161 @@ def test_migration_4_no_own_endpoint(rotkehlchen_api_server):
+            UNION ALL /* using union all as there can be no duplicates so no need to handle them */
+            SELECT base1 AS base_asset, quote1 AS quote_asset, amount1 AS amount, rate1 AS rate, txhash, logindex, time, location FROM C2
+            UNION ALL /* using union all as there can be no duplicates so no need to handle them */
+            SELECT base1 AS base_asset, quote1 AS quote_asset, amount1 AS amount, rate1 AS rate, txhash, logindex, time, location FROM C3
+            UNION ALL /* using union all as there can be no duplicates so no need to handle them */
+            SELECT base1 AS base_asset, quote1 AS quote_asset, amount1 AS amount, rate1 AS rate, txhash, logindex, time, location FROM C4
+            UNION ALL /* using union all as there can be no duplicates so no need to handle them */
+            SELECT base1 AS base_asset, quote1 AS quote_asset, amount1 AS amount, rate1 AS rate, txhash, logindex, time, location FROM C5
+            UNION ALL /* using union all as there can be no duplicates so no need to handle them */
+            SELECT base2 AS base_asset, quote2 AS quote_asset, amount2 AS amount, rate2 AS rate, txhash, logindex, time, location FROM C5
+            UNION ALL /* using union all as there can be no duplicates so no need to handle them */
+            SELECT base1 AS base_asset, quote1 AS quote_asset, amount1 AS amount, rate1 AS rate, txhash, logindex, time, location FROM C6
+            UNION ALL /* using union all as there can be no duplicates so no need to handle them */
+            SELECT base2 AS base_asset, quote2 AS quote_asset, amount2 AS amount, rate2 AS rate, txhash, logindex, time, location FROM C6
+        )
+        SELECT
+            txhash + logindex AS id,
+            time,
+            location,
+            base_asset,
+            quote_asset,
+            'A' AS type, /* always a BUY */
+            amount,
+            rate,
+            NULL AS fee, /* no fee */
+            NULL AS fee_currency, /* no fee */
+            txhash AS link,
+            NULL AS notes /* no notes */
+        FROM SWAPS
+        UNION ALL /* using union all as there can be no duplicates so no need to handle them */
+        SELECT * from trades
+        ;
+        """)  # noqa: E501
+        database.add_amm_swaps(cursor, swaps)
+
+    # verify that the problem is there
+    trades, _ = rotki.events_historian.query_trades(
+        filter_query=TradesFilterQuery.make(),
+        only_cache=True,
+    )
+    assert len(trades) == 1
+    with pytest.raises(UnicodeDecodeError):
+        TradeSchema().dump(trades[0])
+
+    with migration_patch:
+        DataMigrationManager(rotki).maybe_migrate_data()
+
+    trades, _ = rotki.events_historian.query_trades(
+        filter_query=TradesFilterQuery.make(),
+        only_cache=True,
+    )
+    TradeSchema().dump(trades[0])
```

Comment:
Also I think you should add a test for this data migration in case the view is missing. Which should be a no-op and without any failure but also WITHOUT creating that view.